core/src/main/java/com/datastrato/gravitino/connector/CatalogInfo.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.connector;
+
+import com.datastrato.gravitino.Audit;
+import com.datastrato.gravitino.Catalog;
+import com.datastrato.gravitino.Namespace;
+import com.datastrato.gravitino.annotation.Evolving;
+import java.util.Map;
+
+/**
+ * CatalogInfo exposes all the information about a catalog to the connector interface.
+ *
+ * <p>This class object will be passed in through {@link CatalogOperations#initialize(Map,
+ * CatalogInfo)}, users can level this object to get the information about the catalog.
+ */
+@Evolving
+public final class CatalogInfo {
+
+  private Long id;
+
+  private String name;
+
+  private Catalog.Type type;
+
+  private String provider;
+
+  private String comment;
+
+  private Map<String, String> properties;
+
+  private Audit auditInfo;
+
+  private Namespace namespace;
+
+  public CatalogInfo(
```

Comment:
Should we implement a builder pattern and set this constructor to private?

Reply:
This class a just a wrapper of `CatalogEntity` to expose to the users, seems not so necessary to have a builder class.